src/foam2dolfinx/open_foam_reader.py

```diff
@@ -126,68 +178,66 @@ def _create_dolfinx_mesh(self):
 
         # Create dolfinx Mesh
         mesh_ufl = ufl.Mesh(self.mesh_vector_element)
-        self.dolfinx_mesh = create_mesh(
-            MPI.COMM_WORLD, self.connectivity, self.OF_mesh.points, mesh_ufl
+        self.dolfinx_mesh[subdomain] = create_mesh(
+            MPI.COMM_WORLD, self.connectivity, self.OF_mesh[subdomain].points, mesh_ufl
         )
 
     def create_dolfinx_function(
-        self, t: Optional[float] = None, name: Optional[str] = "U"
+        self, t: float, name: str = "U", subdomain: Optional[str] = None
     ) -> dolfinx.fem.Function:
         """Creates a dolfinx.fem.Function from the OpenFOAM file.
 
         Args:
             t: timestamp of the data to read
             name: Name of the field in the openfoam file, defaults to "U" for velocity
+            subdomain: Name of the subdmain in the openfoam file, from which a field is
+                extracted
 
         Returns:
             the dolfinx function
         """
-        self._read_with_pyvista(t=t)
 
-        # Create dolfinx mesh if it doesn't exist
-        if not hasattr(self, "dolfinx_mesh"):
-            self._create_dolfinx_mesh()
+        # read the openfoam data in the filename provided
+        self._read_with_pyvista(t=t, subdomain=subdomain)
+
+        # create the dolfinx mesh
+        self._create_dolfinx_mesh(subdomain=subdomain)
+
+        mesh = self.dolfinx_mesh[subdomain]
```

Comment:
```suggestion
        # create the dolfinx mesh
        if not subdomain in self.dolfinx_mesh:
            self._create_dolfinx_mesh(subdomain=subdomain)

        mesh = self.dolfinx_mesh[subdomain]
```

What's the point of all this otherwise?